include/revng/BasicAnalyses/GeneratedCodeBasicInfo.h

```diff
@@ -414,3 +443,31 @@ struct BlackListTrait<const GeneratedCodeBasicInfo &, llvm::BasicBlock *>
     return !this->Obj.isTranslated(Value);
   }
 };
+
+/// Analysis to compute GCBI.
+class GeneratedCodeBasicInfoAnalysis
+  : public llvm::AnalysisInfoMixin<GeneratedCodeBasicInfoAnalysis> {
+  friend llvm::AnalysisInfoMixin<GeneratedCodeBasicInfoAnalysis>;
+  static llvm::AnalysisKey Key;
+
+public:
+  using Result = GeneratedCodeBasicInfo;
+  Result run(llvm::Module &M, llvm::ModuleAnalysisManager &);
+  Result run(llvm::Function &F, llvm::FunctionAnalysisManager &);
+};
+
+/// Legacy pass manager pass to access GCBI.
+class GeneratedCodeBasicInfoWrapperPass : public llvm::ModulePass {
+  std::unique_ptr<GeneratedCodeBasicInfo> GCBI;
+public:
+  static char ID;
+
+  GeneratedCodeBasicInfoWrapperPass()
+    : llvm::ModulePass(ID) { }
+
+  GeneratedCodeBasicInfo &getGCBI() { return *GCBI; }
+  const GeneratedCodeBasicInfo &getGCBI() const { return *GCBI; }
```

Comment:
Is this needed? Let's put this only if needed.

Reply:
That's needed, it actually provides a way to get a reference to the detached `GeneratedCodeBasicInfo` (EnforceABI should take a const ref IIRC).

Reply:
This doesn't sound right. Do you ever get (from LLVM) a const reference to `GeneratedCodeBasicInfoWrapperPass`?
If not, drop this.

I see no reason why you should pass around `GeneratedCodeBasicInfoWrapperPass`, just pass `GeneratedCodeBasicInfo` to a function that expects a `const &` and it will be converted automatically.

Reply:
Removed.